effekt/jvm/src/main/scala/effekt/Driver.scala

```diff
@@ -100,5 +100,9 @@ trait Driver extends kiama.util.Compiler[EffektConfig, EffektError] { outer =>
   def afterCompilation(source: Source, config: EffektConfig)(implicit C: Context): Unit = {
     // report messages
     report(source, C.messaging.buffer, config)
+    
+    // exit with non-zero code if not in repl/server mode and messaging buffer contains errors
+    if (!(config.repl() || config.server()) && C.messaging.hasErrors)
```

Comment:
Nitpick: I think it might be a bit easier to read this if it were 
```suggestion
    if (config.shouldExitOnError() && C.messaging.hasErrors)
```

with 
```scala
def shouldExitOnError(): Boolean = !repl() && !server()
```
in `EffektConfig.scala` (perhaps with a better name 🤔)

(Similarly to `requiresCompilation()` in `EffektConfig.scala`)